static/js/navigation.js

```diff
@@ -1,269 +1,375 @@
+ * Handle the state of the search
+ */
+function toggleSearch() {
+  const isOpen = navigationContainer.classList.contains("has-search-open");
+
+  if (!isOpen) {
+    closeAll();
+    openSearch();
+  } else {
+    closeAll();
   }
 }
 
-// Careers navigation
-if (nav) {
-  nav.classList.remove("u-hide");
+/**
+ * Resets the state of the search bar
+ */
+function closeSearch() {
+  const searchToggle = document.querySelector(".js-search-button");
+  searchToggle.removeAttribute("aria-pressed");
+  navigationContainer.classList.remove("has-search-open");
+  secondaryNav?.classList.remove("has-search-open");
 }
 
 /**
- * Close rest of navigation (override standard functionality)
- * if in the future we change rest of nav into meganavs
- * remove this function
+ * Opens the search
  */
-function closeAllSimpleNavigation() {
-  const containers = document.querySelectorAll(
-    ".p-navigation__item--dropdown-toggle"
+function openSearch() {
+  const searchToggle = document.querySelector(".js-search-button");
+  const isReduced = navigationContainer.classList.contains(
+    "p-navigation--reduced"
   );
-  containers.forEach((item) => {
-    item.classList.remove("is-active");
-    const dropdown = item.querySelector(".p-navigation__dropdown");
-    dropdown.setAttribute("aria-hidden", "true");
-    if (overlay) {
-      overlay.classList.remove("is-applied");
-    }
-  });
+  searchToggle.setAttribute("aria-pressed", "true");
+  secondaryNav?.classList.add("has-search-open");
+  navigationContainer.classList.add("has-search-open");
 }
 
 /**
- * Close all meganavs
+ * Reset the state of everything
  */
-function closeAllMeganavs() {
-  navDropdowns.forEach((item) => {
-    if (item.classList.contains("is-active")) {
-      item.classList.remove("is-active");
-      dropdownWindow.classList.add("u-animation--slide-dropdown");
-      if (overlay) {
-        overlay.classList.remove("is-applied");
-      }
-    }
-  });
+function closeAll() {
+  closeSearch();
+  closeSecondaryNavigation();
 }
 
-// This is in place to extend future c.com meganavs
-navDropdowns.forEach(function (dropdown) {
-  dropdown.addEventListener("click", function (event) {
-    event.preventDefault();
-
-    var clickedDropdown = this;
-
-    dropdownWindow.classList.remove("u-animation--slide-dropdown");
-    if (overlay && overlay.classList.contains("is-applied")) {
-      overlay.classList.remove("is-applied");
-    }
-
-    navDropdowns.forEach(function (dropdown) {
-      if (dropdown === clickedDropdown) {
-        if (dropdown.classList.contains("is-active")) {
-          closeMenu(dropdown);
-        } else {
-          closeAllSimpleNavigation();
-          dropdown.classList.add("is-active");
-          if (overlay) {
-            overlay.classList.add("is-applied");
-          }
-        }
-      } else {
-        dropdown.classList.remove("is-active");
-        overlay.classList.remove("is-applied");
-      }
-    });
-  });
-});
+// // Careers meganav variables
```

Comment:
Can you add comments on why this piece of code is commented out? Or just remove it?

Reply:
I was keeping it for practicality as I want to use some of it. But I suppose I can just use whats on `main' instead. Removed.